examples/gno.land/r/demo/groups/group.gno

```diff
@@ -0,0 +1,131 @@
+package groups
+
+import (
+	"gno.land/p/demo/avl"
+	"std"
+	"strconv"
+	"time"
+)
+
+type GroupID uint64
+type MemberID uint64
+
+func (gid GroupID) String() string {
+	return strconv.Itoa(int(gid))
+}
+
+func (mid MemberID) String() string {
+	return strconv.Itoa(int(mid))
+}
+
+type Member struct {
+	id        MemberID
+	address   string
+	weight    int
+	metadata  string
+	createdAt time.Time
+}
+
+type Group struct {
+	id           GroupID
+	url          string
+	name         string
+	lastMemberID MemberID
+	members      *avl.MutTree
+	creator      std.Address
+	createdAt    time.Time
+}
+
+func newGroup(id GroupID, url string, name string, creator std.Address) *Group {
+	if !reName.MatchString(name) {
+		panic("invalid name: " + name)
+	}
+	exists := gGroupsByName.Has(name)
+	if exists {
+		panic("Group with such name already exists")
+	}
+	return &Group{
+		id:        id,
+		url:       url,
+		name:      name,
+		creator:   creator,
+		members:   avl.NewMutTree(),
+		createdAt: time.Now(),
+	}
+}
+
+func newMember(id MemberID, gid GroupID, address string, weight int, metadata string) *Member {
```

Comment:
maybe `address` can be of type `std.Address` instead of `string`

Reply:
done.